python/hsfs/engine/python.py

```diff
@@ -872,11 +872,13 @@ def _write_dataframe_kafka(
         writer = self._get_encoder_func(feature_group._get_encoded_avro_schema())
 
         def acked(err, msg):
-            if err is not None and offline_write_options.get("debug_kafka", False):
-                print("Failed to deliver message: %s: %s" % (str(msg), str(err)))
-            else:
-                # update progress bar for each msg
-                progress_bar.update()
+            if err is not None:
+                if err.code() == KafkaError.TOPIC_AUTHORIZATION_FAILED:
+                    raise err  # Stop producing, the user is not authorized
+                if offline_write_options.get("debug_kafka", False):
+                    print("Failed to deliver message: %s: %s" % (str(msg), str(err)))
```

Comment:
This should be before, otherwise it will never be printed, since you reraise the exception.

Reply:
ok we decided to swap the order